src/static_array.cr

```diff
@@ -163,6 +163,25 @@ struct StaticArray(T, N)
     self
   end
 
+  def map
+    clone.map! { |e| yield e }
```

Comment:
Instead of clone, wouldn't be better to create the array with the final values using https://crystal-lang.org/api/0.19.2/StaticArray.html#new%28%26block%3AInt32-%3ET%29-class-method ?

This comment applies for `#map_with_index`'s and `#reverse`.

I am not sure how much performance improvement would be though. You can measure performance using [Benchmark.ips] (https://crystal-lang.org/api/0.19.2/Benchmark.html)